### DISCUSSION TITLE
[SHOW] Handling DateTime and Timestamp with Local Timezone

### DISCUSSION THREAD

**Discussion by darkterminal** [prisma/prisma#15329] [Q&A]
I have been trying for 3 days to make the prisma client understand what I want to enter into the database, namely time (DateTime / Timestamp)

**Api Access Model Example**
```javascript
model api_access {
  api_id       Int       @id @default(autoincrement())
  api_key      String    @db.VarChar(225)
  domain_lists String    @db.Text
  deleted_at   DateTime? @db.Date
  created_at   DateTime  @default(now()) @db.Timestamp(0)
  updated_at   DateTime  @default(now()) @db.Timestamp(0)
}
```

So far prism will do automatic input to the default field using DateTime / Timestamp with UTC date (which may not match the existence of the application created).

Then how to take the Local Timezone and make the prisma understand the unit of time that corresponds to our time zone?

#### 1. Create function that allow you to get local UTC time:
```javascript
// Utils.js
// Usage:
//       let todayISO = new Date().toISOString()
//       let today = getUTCTime(todayISO)

export const getUTCTime = (dateTimeString) => {
    const dt = new Date(dateTimeString);
    const dtNumber = dt.getTime();
    const dtOffset = dt.getTimezoneOffset() * 60000;
    const dtUTC = new Date();
    dtUTC.setTime(dtNumber - dtOffset);

    return dtUTC;
}
```

#### 2. And every-time you want to insert/create data to MySQL Database you can follow this step
```javascript
let todayISO = new Date().toISOString()
let today = getUTCTime(todayISO)

const createApiAccess = await db.api_access.create({
    data: {
        api_key: API_GENERATED,
        domain_lists: 'abc.com,def.com',
        created_at: today,
        updated_at: today
    }
})
```

_the same thing can be done when you want to update data and change the time the data was changed._

Here is my experience using Prisma, I hope that prisma can support a better and more precise time format when used in various time zones and can follow the time zone in the database.

**Comment by nurul3101** [comment]
Hey @darkterminal 👋 

We have this GitHub Issue: #5051 which discusses improving Timezone support for MySQL Databases configured with non UTC Timezones, it's in our radar.

**Reply by 814096703** [reply]
2023 still discussing

**Reply by clockwork53** [reply]
2025 :|

**Reply by Even1s** [reply]
2026 :/

**Comment by pLinho** [comment]
The time zone is set in the system kernel. I think your problem is earlier. An application cannot control its time zone. Your application needs to know how to handle the time zone on each node.

https://www.youtube.com/watch?v=-5wpm-gesOY

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by 7TIN** [comment]
prisma was keep changing my date to one day earlier ( 2025-08-20) to (2025-08-19) 
it was so frustrating i had to use brute force 
like this 

```
function addOneDay(dateStr: string): Date {
  const d = new Date(dateStr);
  d.setDate(d.getDate() + 1);
  return d;
}

```
